src/core/thread/mle.cpp

```diff
@@ -1752,8 +1762,19 @@ ThreadError Mle::HandleDataResponse(const Message &aMessage, const Ip6::MessageI
                                 (mDeviceMode & ModeTlv::kModeFullNetworkData) == 0,
                                 networkData.GetNetworkData(), networkData.GetLength());
 
+    diff = 0;
+
 exit:
     (void)aMessageInfo;
+
+    if (diff > 0)
```

Comment:
Yes. I think that is better.
Here I only try to follow what section 5.15.3 says. 
Not sure my understanding of it is correct or not.